app/workers/preparsing.rb

```diff
@@ -112,19 +112,14 @@ def perform(genotype_id)
 
     # not proper file!
     if not file_is_ok
+      genotype.update!(parse_status: 'error')
       if file_is_duplicate
         UserMailer.duplicate_file(genotype.user_id).deliver_later
-        system("rm #{Rails.root}/public/data/#{genotype.fs_filename}")
-        Genotype.find_by_id(genotype.id).delete
       elsif file_has_mails
         UserMailer.file_has_mails(genotype.user_id).deliver
-        system("rm #{Rails.root}/public/data/#{genotype.fs_filename}")
-        Genotype.find_by_id(genotype.id).delete
       else
         UserMailer.parsing_error(genotype.user_id).deliver_later
         logger.info "file is not ok, sending email"
-        system("rm #{Rails.root}/public/data/#{genotype.fs_filename}")
-        Genotype.find_by_id(genotype.id).delete
```

Comment:
I could use some feedback on this. If we delete the genotype on error, we cannot show the status to the user. What do we want to do as an alternative?

a) We could just not delete it.
b) We could delete it after a period of time and tell the user on the genotypes tab in their settings that we do.
c) Still delete it.
d) Any other ideas?

Reply:
What would the status display on error? Just nothing, or an 'error' thing? Ideally I'd like to delete it but we're not particularly low on space

Reply:
So far it says "error". I was thinkin of adding more information, since sometimes we know what went wrong.

Reply:
Regarding this.. I'm not 100% familiar with how the genotypes are set up in the database, but is it possible to:
Delete the genotype file
Delete most of the genotype data in the db, but leave behind the status field value and any other bare necessities required for displaying status to the user?

Reply:
Yep, that should be easily possible. The files themselves are attached to the `Genotype` model through [paperclip](https://github.com/thoughtbot/paperclip). [The code for our model is here](https://github.com/openSNP/snpr/blob/master/app/models/genotype.rb). 

With paperclip you can just set the `genotype` attribute of a `Genotype` object to `nil` and save the object. This should delete the file while keeping the model: 

```
@genotype = Genotype.find(1)
@genotype.genotype = nil
@genotype.save
```